src/wp-admin/includes/dashboard.php

```diff
@@ -1517,41 +1567,278 @@ function wp_dashboard_empty() {}
+				<?php _e(
+					'Do you have any feedback about this version of ClassicPress?'
+				); ?>
+			</p>
+			<p>
+				<?php printf(
+					__(
+						/* translators: 1: link with instructions to join ClassicPress Slack, 2: link to Support channel, 3: link to Testing channel */
+						'For support, suggestions for improvement, or general discussion about ClassicPress, <a href="%1$s">join our Slack group</a> and send us a message in the <a href="%2s"><strong>#support</strong></a> or <a href="%3s"><strong>#testing</strong></a> channels.'
+					),
+					'https://www.classicpress.net/join-slack/', 'https://classicpress.slack.com/messages/support/', 'https://classicpress.slack.com/messages/testing/'
+				); ?>
+			</p>
+			<p>
+				<?php printf(
+					__(
+						/* translators: link to create a new GitHub issue for this plugin */
+						'For <strong>specific</strong> bug reports or suggestions, <a href="%s">add a new issue on GitHub</a>.',
+						'switch-to-classicpress'
+					),
+					'https://github.com/ClassicPress/ClassicPress/issues/new'
+				); ?>
+			</p>
+
+	</div>
+	<?php		
+}
+
+/**
+ * Returns HTML Mark up for the Petitions Dashboard Widget
+ *
+ * @since CP-1.0.0
+ */
+
+
+function cp_dashboard_petitions_table_body ( $votesCount, $link, $title, $author, $status, $createdTime ) {
+?>
+	<tr>
+		<td class="votes-count"><?php echo esc_attr( $votesCount ); ?></td>
+
+		<td class="petition">
+			<a target="_blank" href="<?php echo esc_url( $link ) ?>"><strong><?php echo esc_attr__( $title )?><span class="screen-reader-text"><?php echo esc_attr__( '(opens in a new window)' ); ?></span><span aria-hidden="true" class="dashicons dashicons-external"></span></strong></a>
+			<?php
+				esc_attr__( 'by' ) . ' ' . ucwords(  esc_attr( $author ) );
+
+				if ( $status == "open" ){
+					echo esc_attr__( ' - ' ) . ' ' . human_time_diff( strtotime( $createdTime ), current_time('timestamp') ) . ' ' . esc_attr__( 'ago' );
+				} 
+				elseif ( $status == "planned" ){
+					echo ' - ' . '<span class="planned">' . esc_attr__( ucfirst( $status ) ) . '</span>';
+				} 
+				else{
+					echo ' - ' . '<span class="started">' . esc_attr__( ucfirst( $status ) ) . '</span>';
+				}
+			?>
+		</td>
+	</tr>
+<?php
 }
+
+/**
+ * Callback function for the petitions dashboard widget
+ *
+ * @since CP-1.0.0
+ */
+function cp_dashboard_petitions() {
+
+	
+	/**
+	 * Query API for JSON data -> decode results to php
+	 *
+	 * @since CP-1.0.0
+	 * @return array
+	 */
+	$api_url = 'https://api-v1.classicpress.net/features/1.0/';
+
+	/**
+	 * Response should be an array with:
+	 *  'most-wanted []' - array() - A user-friendly platform name, if it can be determined
+	 *  'trending []' - array() - A user-friendly browser name
+	 *  'recent []' - array() - The version of the browser the user is using
+	 *  'tags [] - array() - The version of the browser the user is using
+	 *  'link - string - The version of the browser the user is using
+	 */
+	$response = wp_remote_get( $api_url );
```

Comment:
Do I correctly understand that this happens on page load?
This probably should be similar to `wp_dashboard_primary()`, which load news using AJAX request and doesn't affect the page loading time.

Reply:
Need to look into this implementation. However, it's on page load

Reply:
This API endpoint is designed to be callable directly from within a browser.

An alternative would be to cache the result of the query in a transient.  However, calling the API from the browser would be preferable because it avoids issues with failed requests and how they interact with caching.

Reply:
I have added a transient key for the JSON data from URL if no prior key is found. It has an expiry after 12 hours.

Reply:
That reduces the API calls and also the load times on page load.